src/api.v2/helpers/pipeline/pipelineConstruct/qcHelpers.js

```diff
@@ -17,13 +20,46 @@ const qcStepsWithFilterSettings = [
 const hasFilteredCellIdsAvailable = async (experimentId) => (
   await fileExists(FILTERED_CELLS, experimentId)
 );
+
+const getCellLevelMetadataFileChanged = async (experimentId) => {
+  const [cellLevelMetadata] = await new CellLevelMeta()
+    .getMetadataByExperimentIds([experimentId]);
+  const { id: currentCellMetadataId = null } = cellLevelMetadata || {};
+
+  const execution = await new ExperimentExecution()
+    .find({ experiment_id: experimentId, pipeline_type: QC_PROCESS_NAME })
+    .first();
+
+  const lastRunCellMetadataId = execution.lastPipelineParams
+    ? execution.lastPipelineParams.cellMetadataId : null;
+
+  return lastRunCellMetadataId !== currentCellMetadataId;
+};
+
 // getFirstQCStep returns which is the first step of the QC to be run
-// processingConfigUpdates is not ordered
-const getFirstQCStep = async (experimentId, processingConfigUpdates, backendCompletedSteps) => {
+// processingConfigUpdatedKeys is not ordered
+const getFirstQCStep = async (
+  experimentId, processingConfigUpdatedKeys, backendCompletedSteps, status,
+) => {
+  if (
+    processingConfigUpdatedKeys.length === 0
+    && await getCellLevelMetadataFileChanged(experimentId)
+  ) {
+    return filterToStepName.configureEmbedding;
+  }
+
+  if (processingConfigUpdatedKeys.length === 0 && status !== FAILED) {
```

Comment:
How come status is needed now? In case of FAILED would it not go to retry directly using the `retry` endpoint? How will it reach this point?

Reply:
The `retry` endpoint is implemented only for gem2s and seurat, not qc. I did try to implement this in the retry endpoint in a previous iteration of this feature but abandoned it